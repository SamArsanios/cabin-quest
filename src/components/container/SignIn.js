/* eslint-disable no-unused-expressions */
/* eslint-disable react/no-access-state-in-setstate */
/* eslint-disable react/forbid-prop-types */
/* eslint-disable react/require-default-props */
/* eslint-disable react/no-did-update-set-state */

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Button, Form } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { authorizeUser } from '../../redux/actions/userActions';
import Errors from '../presentation/Errors';
import Loading from '../presentation/Loading';

class SignIn extends Component {
  constructor(props) {
    super(props);
    this.state = {
      isSubmit: false,
      err: false,
      username: '',
      email: '',
      password: '',
    };
  }

  componentDidUpdate() {
    const {
      loggedIn, username, history, errors,
    } = this.props;
    const jwt = localStorage.getItem('jwt');
    const { isSubmit } = this.state;
    isSubmit
    && errors
    && this.setState({
      ...this.state,
      isSubmit: false,
    });
    jwt && loggedIn && history.push(`/user/${username}`);
  }

  handleSubmit = (e) => {
    const {
      authorizeUser,
      currentUser,
      loggedIn,
      history,
    } = this.props;

    e.preventDefault();
    this.setState({
      isSubmit: true,
    });

    authorizeUser(this.state);
    currentUser
        && loggedIn
        && history.push(`/user/${currentUser.username}`);
  };

  handleChange = (e) => {
    const { id, value } = e.target;
    this.setState({
      ...this.state,
      [id]: value,
    });
  };

  render() {
    const { errors } = this.props;

    const { isSubmit } = this.state;

    return (
      <div className="signin auth">

        <div className="auth-header-container">
          {errors && (
          <div className="loading">
            <Errors />
          </div>
          )}

          <h1 className="auth-header py-2 text-center font-weight-bolder">
            Sign In
          </h1>
        </div>
        <Form
          className="user-form px-5 py-4 shadow-lg bg-white"
          onSubmit={this.handleSubmit}
        >
          <Form.Group controlId="username" className="pb-3">
            <Form.Control
              required
              type="username"
              placeholder="Enter Username"
              onChange={this.handleChange}
            />
          </Form.Group>

          <Form.Group controlId="email" className="pb-3">
            <Form.Control
              required
              type="email"
              placeholder="Enter email"
              onChange={this.handleChange}
            />
            {/* <Form.Text className="text-muted">
              We will never share your email with anyone else.
            </Form.Text> */}
          </Form.Group>

          <Form.Group controlId="password" className="pb-5">
            <Form.Control
              required
              type="password"
              placeholder="Password"
              onChange={this.handleChange}
            />
          </Form.Group>

          {isSubmit && !errors && (
          <div className="loading">
            <Loading />
          </div>
          )}

          <Button className="btn hero-btn w-100" type="submit">
            Sign In
          </Button>
          <p className="text-center mt-3 font-weight-bolder auth-text">OR</p>
          <Link to="/signup" className="my-3 text-center w-100">
            {/* {' '} */}
            <Button className="btn hero-btn w-100">
              Sign Up
            </Button>
          </Link>
        </Form>
      </div>
    );
  }
}

SignIn.propTypes = {
  errors: PropTypes.any,
  loggedIn: PropTypes.any,
  username: PropTypes.string.isRequired,
  currentUser: PropTypes.any,
  authorizeUser: PropTypes.func.isRequired,
  history: PropTypes.any,
};

const mapStateToProps = (state) => ({
  errors: state.error.err,
  username: state.userData.username,
  currentUser: state.userData.currentUser,
  loggedIn: state.userData.loggedIn,
});

export default connect(mapStateToProps, { authorizeUser })(SignIn);
