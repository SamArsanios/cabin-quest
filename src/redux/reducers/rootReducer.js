import { combineReducers } from 'redux';
import { errorReducer } from './utilityReducer';
import cabinReducer from './cabinReducer';
import { successReducer } from './utilityReducer';
import userReducer from './userReducer';

const rootReducer = combineReducers({
  data: cabinReducer,
  error: errorReducer,
  userData: userReducer,
  succMsg: successReducer,
});

export default rootReducer;
