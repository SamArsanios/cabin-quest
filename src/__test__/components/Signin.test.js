import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import { composeWithDevTools } from 'redux-devtools-extension';
import SignIn from '../../components/container/SignIn';
import rootReducer from '../../redux/reducers/rootReducer';

const state = {};
const middleware = [thunk];
const store = createStore(rootReducer, state, composeWithDevTools(applyMiddleware(...middleware)));
const Wrapper = ({ children }) => (
  <Provider store={store}>{children}</Provider>
);

Wrapper.propTypes = {
  children: PropTypes.shape({}).isRequired,
};

describe('SignIn', () => {
  it('renders correctly', () => {
    const tree = render(<SignIn />, { wrapper: Wrapper });
    expect(tree).toMatchSnapshot();
  });
});
