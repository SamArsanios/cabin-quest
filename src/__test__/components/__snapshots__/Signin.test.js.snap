// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="signin auth"
      >
        <div
          class="auth-header-container"
        >
          
          <h1
            class="auth-header py-2 text-center font-weight-bolder"
          >
            Sign In
          </h1>
        </div>
        <form
          class="user-form px-5 py-4 shadow-lg bg-white"
        >
          <div
            class="pb-3 form-group"
          >
            <input
              class="form-control"
              id="username"
              placeholder="Enter Username"
              required=""
              type="username"
            />
          </div>
          <div
            class="pb-3 form-group"
          >
            <input
              class="form-control"
              id="email"
              placeholder="Enter email"
              required=""
              type="email"
            />
          </div>
          <div
            class="pb-5 form-group"
          >
            <input
              class="form-control"
              id="password"
              placeholder="Password"
              required=""
              type="password"
            />
          </div>
          <button
            class="btn hero-btn w-100 btn btn-primary"
            type="submit"
          >
            Sign In
          </button>
          <p
            class="text-center mt-3 font-weight-bolder auth-text"
          >
            OR
          </p>
          <a
            class="my-3 text-center w-100"
            href="/signup"
          >
            <button
              class="btn hero-btn w-100 btn btn-primary"
              type="button"
            >
              Sign Up
            </button>
          </a>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="signin auth"
    >
      <div
        class="auth-header-container"
      >
        
        <h1
          class="auth-header py-2 text-center font-weight-bolder"
        >
          Sign In
        </h1>
      </div>
      <form
        class="user-form px-5 py-4 shadow-lg bg-white"
      >
        <div
          class="pb-3 form-group"
        >
          <input
            class="form-control"
            id="username"
            placeholder="Enter Username"
            required=""
            type="username"
          />
        </div>
        <div
          class="pb-3 form-group"
        >
          <input
            class="form-control"
            id="email"
            placeholder="Enter email"
            required=""
            type="email"
          />
        </div>
        <div
          class="pb-5 form-group"
        >
          <input
            class="form-control"
            id="password"
            placeholder="Password"
            required=""
            type="password"
          />
        </div>
        <button
          class="btn hero-btn w-100 btn btn-primary"
          type="submit"
        >
          Sign In
        </button>
        <p
          class="text-center mt-3 font-weight-bolder auth-text"
        >
          OR
        </p>
        <a
          class="my-3 text-center w-100"
          href="/signup"
        >
          <button
            class="btn hero-btn w-100 btn btn-primary"
            type="button"
          >
            Sign Up
          </button>
        </a>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
